editoast/src/models/paced_train.rs

```diff
@@ -57,6 +57,58 @@ pub struct PacedTrain {
 }
 
 impl PacedTrain {
+    pub fn apply_exceptions(&self, exceptions: &[PacedTrainException]) -> Vec<TrainSchedule> {
+        exceptions
+            .iter()
+            .map(|exception| self.apply_exception(exception))
+            .collect()
+    }
+
+    pub fn apply_exception(&self, exception: &PacedTrainException) -> TrainSchedule {
+        let mut train_schedule = Self::into_first_occurrence(self.clone());
```

Comment:
Isn't the following not working?
```suggestion
        let mut train_schedule = self.clone().into_first_occurrence();
```